source/common/common/base64.cc

```diff
@@ -143,7 +143,7 @@ inline void encodeLast(uint64_t pos, uint8_t last_char, std::string& ret,
 
 } // namespace
 
-std::string Base64::decode(const std::string& input) {
+std::string Base64::decode(absl::string_view input) {
```

Comment:
I think preserving const-ness won't do much harm, but will help to detect erroneous mutation of `input` inside the function.
```suggestion
std::string Base64::decode(const absl::string_view input) {
```
Here and in the other functions.

Reply:
I don't feel that strongly about this, but we don't tend to do this from what I've seen in the Envoy codebase. Also why would we do this for `absl::string_view` and not for (say) `int` on input params?

Reply:
top level const shouldn't be in declaration because it doesn't affect method signature (https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html)
but in implementation it is encouraged.

Reply:
Thank you @lizan! I stand corrected )

Reply:
thank y'all got a great insight!!